lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -85,13 +85,67 @@ class AwsCompileCloudFrontEvents {
       properties: {
         AllowedMethods: {
           oneOf: [
-            { enum: ['GET', 'HEAD'] },
-            { enum: ['GET', 'HEAD', 'OPTIONS'] },
-            { enum: ['GET', 'HEAD', 'OPTIONS', 'PUT', 'PATCH', 'POST', 'DELETE'] },
+            {
+              uniqueItems: true,
+              minItems: 2,
+              items: { enum: ['GET', 'HEAD'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 3,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 7,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS', 'PUT', 'PATCH', 'POST', 'DELETE'] },
+            },
           ],
         },
         CachedMethods: {
-          oneOf: [{ enum: ['GET', 'HEAD'] }, { enum: ['GET', 'HEAD', 'OPTIONS'] }],
+          oneOf: [
+            {
+              uniqueItems: true,
+              minItems: 2,
+              items: { enum: ['GET', 'HEAD'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 3,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS'] },
+            },
+          ],
+        },
+        ForwardedValues: {
```

Comment:
Hmm.. sorry for late call, but I've just read  in AWS doc, that this field is deprecated, and so far we resisted adding schema for such properties.

Do you feel it's important to ensure recognition of it? I see that we're anyway introducing this field, so maybe it appears needed simply to reset default as introduced by us (?)

Reply:
Valid point! As you pointed out, I think it's good to have it there at least while it's used internally so it can be overridden. The option is still available in the AWS console, but as a legacy setting. 

The new way seems to be to create a custom [Cache Policy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudfront-cachepolicy.html) which I don't think Serverless is supporting yet, and probably is semi complicated to get working properly. I guess it needs to be a separate Resource and then use the id from the new Cache Policy etc. Feels like another PR maybe :)

What do you think?

Reply:
Ok, let's leave support for this (so there's a mean to override what internals put in)

And let's leave switching to Cache Policy to other PR. I've created dedicated issue: https://github.com/serverless/serverless/issues/8317